packages/keyvault/lib/objectIdentifier.ts

```diff
@@ -0,0 +1,236 @@
+    return ObjectIdentifier.parseObjectIdentifier("secrets", identifier);
+};
+
+/** Creates an ObjectIdentifier object for a certificate.
+ * @param {string} vault The vault URI.
+ * @param {string} name The certificate name.
+ * @param {string} [version=undefined] The object version.
+ * @return {ObjectIdentifier} An object that represents the certificate identifier.
+ */
+export function createCertificateIdentifier(vault: string, name: string, version: string): ObjectIdentifier {
+    return ObjectIdentifier.createObjectIdentifier("certificates", vault, name, version);
+};
+
+/** Parses a string containing a certificate identifier and returns the ObjectIdentifier object.
+ * @param {string} identifier The certificate identifier (an URI).
+ * @return {ObjectIdentifier} An object that represents the certificate identifier.
+ */
+export function parseCertificateIdentifier(identifier: string): ObjectIdentifier {
+    return ObjectIdentifier.parseObjectIdentifier("certificates", identifier);
+};
+
+/** Creates an ObjectIdentifier object for a certificate operation.
+ * @param {string} vault The vault URI.
+ * @param {string} name The certificate name.
+ * @return {ObjectIdentifier} An object that represents the certificate identifier.
+ */
+export function createCertificateOperationIdentifier(vault: string, name: string): ObjectIdentifier {
+    var objId = ObjectIdentifier.createObjectIdentifier("certificates", vault, name, "pending");
+    objId.baseIdentifier = objId.identifier;
+    objId.version = undefined;
+    return objId;
+};
+
+/** Parses a string containing a certificate identifier and returns the ObjectIdentifier object.
+ * @param {string} identifier The certificate identifier (an URI).
+ * @return {ObjectIdentifier} An object that represents the certificate identifier.
+ */
+export function parseCertificateOperationIdentifier(identifier: string): ObjectIdentifier {
+    var objId = ObjectIdentifier.parseObjectIdentifier("certificates", identifier);
+    objId.baseIdentifier = objId.identifier;
+    objId.version = undefined;
+    return objId;
+};
+
+/** Creates an ObjectIdentifier object for a certificate issuer.
+ @param {string} vault The vault URI.
+ @param {string} name The certificate issuer name.
+ @return {ObjectIdentifier} An object that represents the certificate issuer identifier.
+*/
+export function createIssuerIdentifier(vault: string, name: string): ObjectIdentifier {
+    return ObjectIdentifier.createObjectIdentifier("certificates/issuers", vault, name);
+};
+
+/** Parses a string containing a certificate issuer identifier and returns the ObjectIdentifier object.
+ @param {string} identifier The certificate issuer identifier (an URI).
+ @return {ObjectIdentifier} An object that represents the certificate issuer identifier.
+*/
+export function parseIssuerIdentifier(identifier: string): ObjectIdentifier {
+    var baseUri;
+    try {
+        baseUri = parse(identifier, true, true);
```

Comment:
We have a URLBuilder that can parse URLs in ms-rest-js. Would it be possible to use that instead to reduce the dependency load?